tracer/src/Datadog.Trace/Ci/TestModule.cs

```diff
@@ -0,0 +1,275 @@
+// <copyright file="TestModule.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using Datadog.Trace.Ci.Tagging;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Vendors.Serilog;
+
+namespace Datadog.Trace.Ci;
+
+/// <summary>
+/// CI Visibility test module
+/// </summary>
+public sealed class TestModule
+{
+    private static readonly AsyncLocal<TestModule?> CurrentModule = new();
+    private readonly Span _span;
+    private readonly Dictionary<string, TestSuite> _suites;
+    private int _finished;
+
+    private TestModule(string name, string? framework = null, string? frameworkVersion = null, DateTimeOffset? startDate = null)
```

Comment:
nit: Same argument, I'm not a fan of default parameters here

Reply:
.ctor are not public, but I did remove the optional parameters from the static methods on [1d3508b](https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/1d3508be7337c2024f64559ad008daeef8f2921f)